File: services/sphana-data-store-service/src/python/sphana_data_store/controllers/indices/v1/schemas/index_details.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel

class IndexDetails(BaseModel):
    index_name: Optional[str] = None
    description: Optional[str] = None
    media_type: Optional[str] = None
    dimension: Optional[int] = None
    number_of_shards: Optional[int] = None
    creation_timestamp: Optional[datetime] = None
    modification_timestamp: Optional[datetime] = None

File: services/Sphana/Services/Sphana.DataStore/Domain/Models/IndexDetails.cs
namespace Sphana.DataStore.Domain.Models;

/// <summary>
/// Represents the metadata details of a vector index.
/// </summary>
public sealed class IndexDetails
{
    public required string IndexName { get; init; }
    public required string Description { get; init; }
    public required string MediaType { get; init; }
    public required int Dimension { get; init; }
    public required int NumberOfShards { get; init; }
    public required DateTime CreationTimestamp { get; init; }
    public required DateTime ModificationTimestamp { get; init; }
}